adapters/AppLovin/AppLovinAdapter/RTB/AppLovinRtbAdapter.m

```diff
@@ -0,0 +1,137 @@
+
++ (void)initialize {
+    [super initialize];
+    
+    ALGlobalSdkSet = [NSMutableSet set];
+}
+
+- (void)dealloc {
+    self.bannerRenderer = nil;
+    self.interstitialRenderer = nil;
+    self.rewardedRenderer = nil;
+}
+
+#pragma mark - GADRTBAdapter
+
++ (void)setUp {}
+
++ (void)updateConfiguration:(GADMediationServerConfiguration *)configuration {
+    //Initialize sdk(s) from configuration
+    for (GADMediationCredentials *credentials in configuration.credentials) {
+        ALSdk *sdk = [GADMAdapterAppLovinUtils retrieveSDKFromCredentials:credentials.settings];
+        [ALGlobalSdkSet addObject:sdk];
+    }
+}
+
++ (GADVersionNumber)adapterVersion {
+    NSString *versionString = GADMAdapterAppLovinConstant.adapterVersion;
+    NSArray *versionComponents = [versionString componentsSeparatedByString:@"."];
+    
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 4) {
+        version.majorVersion = [versionComponents[0] integerValue];
+        version.minorVersion = [versionComponents[1] integerValue];
+        // Adapter versions have 2 patch versions. Multiply the first patch by 100.
+        version.patchVersion = [versionComponents[2] integerValue] * 100 + [versionComponents[3] integerValue];
+    }
+    return version;
+}
+
++ (GADVersionNumber)adSDKVersion {
+    NSString *versionString = ALSdk.version;
+    NSArray *versionComponents = [versionString componentsSeparatedByString:@"."];
+    
+    GADVersionNumber version = {0};
+    if (versionComponents.count == 3) {
+        version.majorVersion = [versionComponents[0] integerValue];
+        version.minorVersion = [versionComponents[1] integerValue];
+        version.patchVersion = [versionComponents[2] integerValue];
+    }
+    return version;
+}
+
++ (Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterAppLovinExtras class];
+}
+
+- (void)collectSignalsForRequestParameters:(nonnull GADMediationRequestParameters *)params
+                         completionHandler:(nonnull GADRTBSignalCompletionHandler)handler {
+    self.signalCompletionHandler = handler;
+    
+    if (params) {
+        [GADMAdapterAppLovinUtils log:@"Extras for signal collection: %@", params.extras];
+    }
+    
+    ALSdk *sdk = [GADMAdapterAppLovinUtils retrieveSDKFromCredentials:params.credentials.settings];
+    NSString *signal = sdk.adService.bidToken;
+    
+    if ( signal.length > 0 ) {
+        [GADMAdapterAppLovinUtils log:@"Generated bid token %@", signal];
+        handler(signal, nil);
+    } else {
+        NSString *errorMessage = @"Failed to generate bid token";
+        NSError *error = [NSError errorWithDomain:GADMAdapterAppLovinConstant.rtbErrorDomain
+                                             code:kGADErrorMediationAdapterError
+                                         userInfo:@{
+                                                    NSLocalizedFailureReasonErrorKey : @"Failed to generate bid token"
```

Comment:
Btw - when I asked to extract the string out, I meant to replace this one as well